sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1232,7 +1236,10 @@ def _bin_data(self, X, is_training_data):
             )
         tic = time()
         if is_training_data:
-            X_binned = self._bin_mapper.fit_transform(X)  # F-aligned array
+            self._bin_mapper = self._bin_mapper.fit(
+                X, sample_weight=sample_weight
+            )  # F-aligned array
+            X_binned = self._bin_mapper.transform(X)
```

Comment:
I think we should be able to keep calling `fit_transform`, even we have weights.

```suggestion
            X_binned = self._bin_mapper.fit_transform(
                X, sample_weight=sample_weight
            )  # F-aligned array
```